internal/container/client.go

```diff
@@ -0,0 +1,55 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package container
+
+import (
+	"github.com/edgexfoundry/go-mod-bootstrap/di"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/coredata"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/general"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/metadata"
+)
+
+var MetadataDeviceClientName = "MetadataDeviceClient"
+var MetadataDeviceServiceClientName = "MetadataDeviceServiceClient"
+var MetadataDeviceProfileClientName = "MetadataDeviceProfileClient"
+var MetadataAddressableClientName = "MetadataAddressableClient"
+var MetadataProvisionWatcherClientName = "MetadataProvisionWatcherClient"
+var MetadataGeneralClientName = "MetadataGeneralClient"
```

Comment:
@tonyespy , it is used as a key in the underlying map the the `dic` uses to manage the items you put into the container

Reply:
For its one and the only usage across SDK: https://github.com/edgexfoundry/device-sdk-go/blob/ce316cea3dee80a2f116041d01b56dbf0103bc38/internal/provision/profiles.go#L155

I also feel like we should rename it to just `GeneralClient`  because it's not part of metadata client according to go-mod-core-contracts: https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/general/client.go

Reply:
@hahattan thanks for the explanation. I agree that `GeneralClient` is a better name, but we probably can't change that for v1.

Reply:
@tonyespy , this code is under `internal` so we can rename it. :-)